### PR TITLE
Debug build fix

### LINKED ISSUE #6529: -DCMAKE_BUILD_TYPE=Debug fails to build
```
$ c++ --version
FreeBSD clang version 8.0.1 (tags/RELEASE_801/final 366581) (based on LLVM 8.0.1)
Target: x86_64-unknown-freebsd13.0
Thread model: posix
InstalledDir: /usr/bin

$ git describe --tags --always
v0.0.7-74-gc1b0934f7

$ cmake -GNinja -DUSE_SYSTEM_FFMPEG=ON -DUSE_SYSTEM_LIBPNG=ON -DCMAKE_BUILD_TYPE=Debug /path/to/rpcs3
$ ninja
[...]
ld: error: undefined symbol: id_manager::id_traits<sys_vm_t, void>::invalid
>>> referenced by sys_vm.cpp:19 (/path/to/rpcs3/rpcs3/Emu/Cell/lv2/sys_vm.cpp:19)
>>>               sys_vm.cpp.o:(sys_vm_t::~sys_vm_t()) in archive rpcs3/Emu/librpcs3_emu.a
```

### PR DESCRIPTION
**Pull request by MSuih**
Fixes #6529

### CODE REVIEW

**Comment by elad335** [comment]
All those `& -` uses are align down, maybe it should be a function instead of doing this workaround everytime.

**Comment by MSuih** [comment]
Dropped the other commit for now, might revisit it later if nobody else does.

File: rpcs3/Emu/IdManager.h [hunk 1]
```diff
@@ -27,10 +27,10 @@ namespace id_manager
...
 	struct id_traits<T, std::void_t<decltype(&T::id_base), decltype(&T::id_step), decltype(&T::id_count)>>
 	{
-		static const u32 base    = T::id_base;
-		static const u32 step    = T::id_step;
-		static const u32 count   = T::id_count;
-		static const u32 invalid = base > 0 ? 0 : -1;
+		static const u32 base        = T::id_base;
+		static const u32 step        = T::id_step;
+		static const u32 count       = T::id_count;
+		static constexpr u32 invalid = base > 0 ? 0 : -1;
```
Comment (Nekotekina) [inline rpcs3/Emu/IdManager.h]: How about
```suggestion
		static const u32 invalid = -!base;
```

That's really interesting...

Comment (elad335) [reply rpcs3/Emu/IdManager.h]: Atleast -u32{!base} to avoid warning.

Comment (MSuih) [reply rpcs3/Emu/IdManager.h]: `-!` doesnt seem to cause any warnings.

Comment (elad335) [reply rpcs3/Emu/IdManager.h]: I would assume the default type path of `u32 = -int(bool(!base))` produces a warning on some compilers, have you checked all compilers?

Comment (MSuih) [reply rpcs3/Emu/IdManager.h]: It triggers `warning C4804:  '-': unsafe use of type 'bool' in operation` for anyone using VS solution files. For cmake this warning has been disabled in ConfigureCompiler.cmake.

Comment (Nekotekina) [reply rpcs3/Emu/IdManager.h]: Well, `-+!base` then. Who can do shorter.

**Comment by KingKingov** [comment]
vcruntime140_1.dll is missing, rpcs3 no work

**Pull request merged**